### PR TITLE
docs: add hashlinks and replace CIDs with multihash

### PR DESCRIPTION
**Pull request by troyronda**
Signed-off-by: Troy Ronda <troy.ronda@securekey.com>

### CODE REVIEW

**Review by fqutishat — APPROVED** [review]

File: index.html [hunk 1]
```diff
@@ -405,40 +433,41 @@ <h2 id="format">The did:orb Format</h2>
...
 
         <pre class="example" title="A canonical Orb DID">
-          did:orb:bafkreiatkubvbkdidscmqynkyls3iqawdqvthi7e6mbky2amuw3inxsi3y:EiDyOQbbZAa3aiRzeCkV7LOx3SERjjH93EXoIM3UoN4oWg
+          did:orb:uEiDlXjleTwr4eZalpXVy086zs-TPK-h54ojbpl7EBvZeHQ:EiDyOQbbZAa3aiRzeCkV7LOx3SERjjH93EXoIM3UoN4oWg
         </pre>
 
-        <pre class="example" title="An Orb DID that uses the WebCAS scheme for content discovery">
-          did:orb:webcas:example.com:bafkreiatkubvbkdidscmqynkyls3iqawdqvthi7e6mbky2amuw3inxsi3y:EiDyOQbbZAa3aiRzeCkV7LOx3SERjjH93EXoIM3UoN4oWg
+        <pre class="example" title="An Orb DID that uses the Hashlink scheme for content discovery">
+          did:orb:hl:uEiDlXjleTwr4eZalpXVy086zs-TPK-h54ojbpl7EBvZeHQ:uoQ-Bc2h0dHBzOi8vZXhhbXBsZS5jb20:EiDyOQbbZAa3aiRzeCkV7LOx3SERjjH93EXoIM3UoN4oWg
```
Comment (troyronda) [inline index.html:459]: We should consider having the multibase base64url prefix (`u`) on the Sidetree suffix too.

File: index.html [hunk 2]
```diff
@@ -405,40 +433,41 @@ <h2 id="format">The did:orb Format</h2>
...
         <pre>
           did-orb-format        = "did:orb:" (orb-canonical-did / orb-scheme-did / unpublished-did)
-          orb-canonical-did     = min-graph-cid ":" did-suffix
-          orb-scheme-did        = (dht-scheme / webcas-scheme) ":" orb-canonical-did
-          unpublished-did       = "interim:" reg-name ":" did-suffix [":" long-form-suffix-data]
-          dht-scheme            = ( "ipfs" )
-          webcas-scheme         = "webcas:" reg-name
+          orb-canonical-did     = min-graph-multihash ":" did-suffix
+          orb-scheme-did        = (hashlink-scheme / hostmeta-scheme)
+          hashlink-scheme       = "hl:" min-graph-multihash ":" graph-hl-metadata ":" did-suffix
```
Comment (troyronda) [inline index.html:439]: Our existing resolver driver logic works by fetching a CAS file and then querying hostmeta that is embedded into the anchor origin.

This commit effectively encodes what we are doing for discovery into the DID string:
- Supply a hashlink to the anchor event/credential object that can be parsed to find the anchor origin for the DID.
- Directly supply the information needed to find the hostmeta file (via https or ipns).

File: index.html [hunk 3]
```diff
@@ -405,40 +433,42 @@ <h2 id="format">The did:orb Format</h2>
...
-        <pre class="example" title="An Orb DID that uses the WebCAS scheme for content discovery">
-          did:orb:webcas:example.com:bafkreiatkubvbkdidscmqynkyls3iqawdqvthi7e6mbky2amuw3inxsi3y:EiDyOQbbZAa3aiRzeCkV7LOx3SERjjH93EXoIM3UoN4oWg
+        <pre class="example" title="An Orb DID that uses the Hashlink scheme for content discovery">
+          did:orb:hl:uEiDlXjleTwr4eZalpXVy086zs-TPK-h54ojbpl7EBvZeHQ:uoQ-Bc2h0dHBzOi8vZXhhbXBsZS5jb20:EiDyOQbbZAa3aiRzeCkV7LOx3SERjjH93EXoIM3UoN4oWg
         </pre>
 
-        <pre class="example" title="An Orb DID that uses the IPFS scheme for content discovery">
-          did:orb:ipfs:bafkreiatkubvbkdidscmqynkyls3iqawdqvthi7e6mbky2amuw3inxsi3y:EiDyOQbbZAa3aiRzeCkV7LOx3SERjjH93EXoIM3UoN4oWg
+        <pre class="example" title="An Orb DID that uses the hostmeta HTTPS scheme for content discovery">
+          did:orb:https:example.com:uEiDlXjleTwr4eZalpXVy086zs-TPK-h54ojbpl7EBvZeHQ:EiDyOQbbZAa3aiRzeCkV7LOx3SERjjH93EXoIM3UoN4oWg
```
Comment (troyronda) [inline index.html:463]: note: we could also choose to prefix as `did:orb:hm:https:` or similar.

**Review by bstasyszyn — APPROVED** [review]

File: index.html (hunk 2 above)
Comment (troyronda) [reply index.html:439]: The hashlink to an anchor event/credential also matches up with https://github.com/trustbloc/activityanchors/pull/6.

File: index.html (hunk 3 above)
Comment (sandrask) [reply index.html:463]: I thought we agreed that we will use hl:multihash:metadata for discovery.

Comment (troyronda) [reply index.html:463]: hl:multihash:metadata is described above.

**Pull request merged**